web/client/epics/maps.js

```diff
@@ -312,14 +314,16 @@ const storeDetailsInfoEpic = (action$, store) =>
 /**
  * Create or update map resource with persistence api
  */
-const mapSaveMapResourceEpic = (action$) =>
+const mapSaveMapResourceEpic = (action$, store) =>
     action$.ofType(SAVE_MAP_RESOURCE)
         .exhaustMap(({resource}) => (!resource.id ? createResource(resource) : updateResource(resource))
             .switchMap((rid) => Rx.Observable.from([
                 ...(resource.id ? [loadMapInfo(rid)] : []),
                 resource.id ? toggleControl('mapSave') : toggleControl('mapSaveAs'),
                 mapSaved(),
-                ...(!resource.id ? [mapCreated(rid, assign({id: rid, canDelete: true, canEdit: true, canCopy: true}, resource.metadata), resource.data)]
+                ...(!resource.id ? [
+                    mapCreated(rid, assign({id: rid, canDelete: true, canEdit: true, canCopy: true}, resource.metadata), resource.data),
+                    push(`/viewer/${mapTypeSelector(store.getState())}/${rid}`)]
```

Comment:
Please, explain me or add a comment about this redirect

Reply:
This is the redirect that was previously in MapSave, it was moved here because it is a more appropriate place for it, and when it was in MapSave it was breaking the new contexts functionality. This change was discussed in the dev chat.

Reply:
ok please add a comment that explains it, thanks

Reply:
We save new map and then redirect to that new map. The behaviour hasn't changed. The code if anything has become clearer. Not sure what explanation should be given here.

Reply:
nevermind